src/psyclone/psyir/symbols/symbol_table.py

```diff
@@ -610,6 +616,33 @@ def check_for_clashes(self, other_table, symbols_to_skip=()):
                         f"the supplied table imports it from Container "
                         f"'{other_sym.interface.container_symbol.name}'.")
                 continue
+
+            if other_sym.is_unresolved and this_sym.is_unresolved:
+                # Both Symbols are unresolved.
+                if shared_wildcard_imports:
+                    # The tables have one or more wildcard imports in common.
+                    # Therefore we assume that the two symbols in fact
+                    # represent the same memory location.
+                    continue
+                # pylint: disable=import-outside-toplevel
+                from psyclone.psyir.nodes import IntrinsicCall
+                try:
+                    # An unresolved symbol representing an intrinisc is fine.
+                    _ = IntrinsicCall.Intrinsic[this_sym.name.upper()]
+                    # Take this opportunity to specialise the symbol(s).
+                    if not isinstance(this_sym, IntrinsicSymbol):
+                        this_sym.specialise(IntrinsicSymbol)
+                    if not isinstance(other_sym, IntrinsicSymbol):
+                        other_sym.specialise(IntrinsicSymbol)
+                    continue
+                except KeyError:
+                    pass
```

Comment:
Good spot, thanks. I've removed this code now.

Reply:
It is still at line 671?

Reply:
I should have been more specific. What I've done is tighten things up so that we *only* do this specialising if there's absolutely no way the symbols can be being brought into scope, in which case they must be intrinsics.